index.html

```diff
@@ -77,67 +78,76 @@ <h2>Examples of Usage</h2>
-        <h3>The <dfn data-link-for="isInputPending">isInputPending</dfn> Method</h3>
-        <p>The <code>isInputPending</code> method returns a value based on the strings listed in <code>inputTypes</code>. If the <code>isInputPending</code> method is called then the user agent MUST run the following steps:</p>
+        <h3>The <dfn>isInputPending</dfn> Method</h3>
+        <p>The <code>isInputPending</code> method returns a value based on the options set listed in <code>isInputPendingOptions</code>. If the <code>isInputPending</code> method is called then the user agent MUST run the following steps:</p>
         <ol>
-          <li>If <code>inputTypes</code> is empty then let <code>inputTypes</code> be an array of <code>SupportedEventTypes</code>.</li>
-          <li>Let <code>pendingResult</code> be false.</li>
-          <li>The user agent MAY at this step set <code>pendingResult</code> to true for any reason.</li>
+          <li>Let <i>pendingResult</i> be false.</li>
+          <li>If <i>isInputPendingOptions</i> is not set then set <i>isInputPendingOptions</i> to be the output from running the <a>IsInputPendingOptions</a> constructor with no input options.</li>
+          <li>The user agent MAY at this step set <i>pendingResult</i> to true for any reason.</li>
           <li>The user agent MAY at this step continue to step 6 for any reason.</li>
-          <li>For each <code>item</code> in <code>inputTypes</code>:<ol>
-            <li>If <code>item</code> is not in <code>SupportedEventTypes</code>, continue to next <code>item</code>.</li>
-            <li>If the <dfn data-cite="ecmascript#surrounding-agent">surrounding agent</dfn>'s <dfn data-cite="html#event-loop">event loop</dfn> has any <dfn data-cite="html#task-queue">task queues</dfn> that contain a <dfn data-cite="html#concept-task">task</dfn> that will <dfn data-cite="dom#concept-event-dispatch">dispatch</dfn> an event, with <dfn data-cite="dom#dom-event-type">type</dfn> attribute equal to <code>item</code>, or if the user agent determines that it MAY soon enqueue such a task, and the dispitached event would have an EventTarget that exists within a window of the same origin, the user agent MAY let <code>pendingResult</code> be true and continue to step 6.</li>
+          <li>If the <dfn data-cite="ecmascript#surrounding-agent">surrounding agent</dfn>'s <dfn data-cite="html#event-loop">event loop</dfn> has any <dfn data-cite="html#task-queue">task queues</dfn> that contain a <dfn data-cite="html#concept-task">task</dfn> that will <dfn data-cite="dom#concept-event-dispatch">dispatch</dfn> an event that is a <dfn data-cite="uievents#idl-uievent">UIEvent</dfn> or <dfn data-cite="touch-events#touchevent-interface">TouchEvent</dfn> or a child of either, or if the user agent determines that it MAY soon enqueue such a task, and the dispitached event would have an EventTarget that exists within a window of the same origin, then the user agent MUST run the following steps:<ol>
```

Comment:
```suggestion
          <li>If the <dfn data-cite="ecmascript#surrounding-agent">surrounding agent</dfn>'s <dfn data-cite="html#event-loop">event loop</dfn> has any <dfn data-cite="html#task-queue">task queues</dfn> that contain a <dfn data-cite="html#concept-task">task</dfn> that will <dfn data-cite="dom#concept-event-dispatch">dispatch</dfn> an event that is a <dfn data-cite="uievents#idl-uievent">UIEvent</dfn> or <dfn data-cite="touch-events#touchevent-interface">TouchEvent</dfn> or a child of either, or if the user agent determines that it MAY soon enqueue such a task, and the dispatched event would have an EventTarget that exists within a window of the same origin, then the user agent MUST run the following steps:<ol>
```